tests/data_context/test_data_context_test_yaml_config_usage_stats.py

```diff
@@ -299,3 +299,58 @@ def test_test_yaml_config_usage_stats_simple_sqlalchemy_datasource_subclass(
     # Confirm that logs do not contain any exceptions or invalid messages
     assert not usage_stats_exceptions_exist(messages=caplog.messages)
     assert not usage_stats_invalid_messages_exist(messages=caplog.messages)
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_rule_based_profiler_emits_valid_usage_stats(
+    mock_emit, caplog, empty_data_context_stats_enabled, test_df, tmp_path_factory
+):
+    context = empty_data_context_stats_enabled
+    yaml_config = """
+    name: my_profiler
+    class_name: RuleBasedProfiler
+    module_name: great_expectations.rule_based_profiler
+    config_version: 1.0
+    variables:
+      integer_type: INTEGER
+      timestamp_type: TIMESTAMP
+      max_user_id: 999999999999
+      min_timestamp: 2004-10-19 10:23:54
+    rules:
+      my_rule_for_user_ids:
+        name: my_rule_for_user_ids
+        class_name: Rule
+        module_name: great_expectations.rule_based_profiler.rule
+        domain_builder:
+          class_name: TableDomainBuilder
+        expectation_configuration_builders:
+          - expectation_type: expect_column_values_to_be_of_type
+            class_name: DefaultExpectationConfigurationBuilder
+    """
+    context.test_yaml_config(
+        yaml_config=yaml_config, name="my_profiler", class_name="Profiler"
+    )
+
+    # Substitute anonymized name since it changes for each run
+    anonymized_name = mock_emit.call_args_list[0][0][0]["event_payload"][
+        "anonymized_name"
+    ]
+    assert mock_emit.call_count == 1
+    assert mock_emit.call_args_list == [
+        mock.call(
+            {
+                "event": "data_context.test_yaml_config",
+                "event_payload": {
+                    "anonymized_name": anonymized_name,
+                    "parent_class": "RuleBasedProfiler",
+                },
+                "success": True,
+            }
+        )
+    ]
```

Comment:
@anthonyburdi is this a good enough payload? Do we want to add more information about the nested types? Or perhaps is this something we'll evolve over time as we finalize this feature?

@alexsherstinsky I know you added a ticket for usage stats so please let me know your thoughts.

Reply:
I think a few things we might want to know are: 1. what keys are being used (variables, rules) and perhaps 2. how many rules are configured. No content - just whether the keys exist and number of rules.

Reply:
sounds good! I'm going to defer to the usage_stats ticket we have to flesh this out but I'll reflect these comments therein!